android/java/org/chromium/chrome/browser/crypto_wallet/util/TokenUtils.java

```diff
@@ -66,10 +67,13 @@ private static BlockchainToken[] filterTokens(NetworkInfo selectedNetwork,
                 default:
                     throw new UnsupportedOperationException("Token type not supported.");
             }
-            return typeFilter || isSameToken(t, nativeAsset) || (keepVisibleOnly && !t.visible);
+            return typeFilter || (includeNativeAsset && isSameToken(t, nativeAsset))
+                    || (keepVisibleOnly && !t.visible);
         });
 
-        arrayTokens.add(0, nativeAsset);
+        if (includeNativeAsset) {
```

Comment:
Why this is necessary? Doesn't arrayTokens already have it when `isSameToken(t, nativeAsset)` is `true`?